feathr_project/feathr/spark_provider/_localspark_submission.py

```diff
@@ -0,0 +1,224 @@
+        logger.info(f"{self.spark_job_num} local spark job(s) in this Launcher, only the latest will be monitored.")
+        logger.info(f"Please check auto generated spark command in {self.cmd_file} and detail logs in {self.log_path}.")
+
+        proc = self.latest_spark_proc
+        start_time = time.time()
+        retry = self.retry
+
+        log_r = open(f'{self.log_path}_{self.spark_job_num-1}.txt' , 'r') 
+        while proc.poll() is None and (((timeout_seconds is None) or (time.time() - start_time < timeout_seconds))):
+            time.sleep(1)
+            try:
+                if retry < 1:
+                    logger.warning(f"Spark job has hang for {self.retry * self.retry_sec} seconds. latest msg is {last_line}. please check {log_r.name}")
+                    if self.clean_up:
+                        self._clean_up()
+                        proc.wait()
+                    break
+                last_line = log_r.readlines()[-1]
+                retry = self.retry
+                if last_line == []:
+                    print("_", end="")
+                else:
+                    print(">", end="")
+                    if last_line.__contains__('Feathr Pyspark job completed'):
+                        logger.info(f"Pyspark job Completed")
+                        proc.terminate()
+            except IndexError as e:
+                print("x", end="")
+                time.sleep(self.retry_sec)
+                retry -= 1
+
+        job_duration = time.time() - start_time
+        log_r.close() 
+
+        if proc.returncode == None:
+            logger.warning(f"Spark job not completed after {timeout_seconds} sec time out setting, please check.")
+            if self.clean_up:
+                self._clean_up()
+                proc.wait()
+                return True
+        elif proc.returncode == 1:
+            logger.warning(f"Spark job is not successful, please check.")
+            return False
+        else:
+            logger.info(f"Spark job {self.latest_spark_proc.pid} finished in: {int(job_duration)} seconds with returncode {proc.returncode}")
+            return True
+
+    def _clean_up(self, proc:Popen = None):
+        logger.warning(f"Terminate the spark job due to as clean_up is set to True.")
+        if not proc:
+            self.latest_spark_proc.terminate()
+        else:
+            proc.terminate()
+
+    def get_status(self) -> str:
+        """Get the status of the job, only a placeholder for local spark"""
+        return self.latest_spark_proc.returncode
+
+    def _init_args(self, job_name:str):
+        args = []
+        args.append('spark-submit')
+        args.append('--master local[*]')
+        args.append('--name %s' % job_name)
+        args.append('--conf "spark.driver.extraClassPath=../target/scala-2.12/classes:jars/config-1.3.4.jar:jars/jackson-dataformat-hocon-1.1.0.jar:jars/jackson-module-caseclass_2.12-1.1.1.jar:jars/mvel2-2.2.8.Final.jar:jars/fastutil-8.1.1.jar"' )
```

Comment:
Clean up the classpath, lots of them (if not all) are from my test script and should not be here.

Reply:
cleaned up all of them and the test runs fine.